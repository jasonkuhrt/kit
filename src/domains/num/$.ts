/**
 * Numeric types and utilities with branded types for mathematical constraints.
 *
 * Provides branded number types (Positive, Negative, Even, Odd, etc.) with
 * runtime validation, mathematical operations, range types, and specialized
 * numeric domains like Complex, Ratio, and BigInt. Includes type guards,
 * ordering, and equivalence utilities.
 */
export * as Num from './$$.js'
