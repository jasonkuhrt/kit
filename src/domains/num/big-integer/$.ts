export * as BigInteger from './$$.ts'
