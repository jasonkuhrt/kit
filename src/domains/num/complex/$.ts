export * as Complex from './$$.js'
