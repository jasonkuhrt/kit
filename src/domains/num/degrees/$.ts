// Export the namespace containing all Degrees operations
export * as Degrees from './$$.js'
