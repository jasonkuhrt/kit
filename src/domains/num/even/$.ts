// Export the namespace containing all Even operations
export * as Even from './$$.ts'
