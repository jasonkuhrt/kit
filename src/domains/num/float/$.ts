// Export the namespace containing all Float operations
export * as Float from './$$.js'
