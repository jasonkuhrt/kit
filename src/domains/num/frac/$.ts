// Export the namespace containing all Frac operations
export * as Frac from './$$.js'
