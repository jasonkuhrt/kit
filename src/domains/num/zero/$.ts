// Export the namespace containing all Zero operations
export * as Zero from './$$.js'
