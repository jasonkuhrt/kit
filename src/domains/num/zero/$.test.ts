import { Test } from '#test'
import { Zero } from './$.js'

Test.describe('is')
  .on(Zero.is)
  .cases(
    // Valid cases - only 0 is zero
    [0],
    [-0], // -0 === 0 in JavaScript
    // Invalid cases - any non-zero value
    [1],
    [-1],
    [0.1],
    [-0.1],
    [Infinity],
    [-Infinity],
    [NaN],
    ['0'],
    [null],
    [undefined],
  )
  .test()

Test.describe('from')
  .on(Zero.from)
  .cases(
    // Valid cases
    [0],
    [-0],
    // Invalid cases - should throw
    [1],
    [-1],
    [0.1],
    [-0.1],
    [Infinity],
    [NaN],
  )
  .test()

Test.describe('tryFrom')
  .on(Zero.tryFrom)
  .cases(
    // Valid cases
    [0],
    [-0],
    // Invalid cases - should return null
    [1],
    [-1],
    [0.1],
    [-0.1],
    [Infinity],
    [NaN],
  )
  .test()
