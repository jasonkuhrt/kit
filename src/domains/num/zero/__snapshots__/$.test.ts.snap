// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`from > from(-0.1) > from(-0.1) 1`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
-0.1
╠══════════════════════════════════════════════════╣ THEN THROWS
Error: Value must be zero, got: -0.1
╚══════════════════════════════════════════════════╝"
`;

exports[`from > from(-1) > from(-1) 1`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
-1
╠══════════════════════════════════════════════════╣ THEN THROWS
Error: Value must be zero, got: -1
╚══════════════════════════════════════════════════╝"
`;

exports[`from > from(0) > from(0) 1`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
0
╠══════════════════════════════════════════════════╣ THEN RETURNS
0
╚══════════════════════════════════════════════════╝"
`;

exports[`from > from(0) > from(0) 2`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
0
╠══════════════════════════════════════════════════╣ THEN RETURNS
0
╚══════════════════════════════════════════════════╝"
`;

exports[`from > from(0.1) > from(0.1) 1`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
0.1
╠══════════════════════════════════════════════════╣ THEN THROWS
Error: Value must be zero, got: 0.1
╚══════════════════════════════════════════════════╝"
`;

exports[`from > from(1) > from(1) 1`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
1
╠══════════════════════════════════════════════════╣ THEN THROWS
Error: Value must be zero, got: 1
╚══════════════════════════════════════════════════╝"
`;

exports[`from > from(null) > from(null) 1`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
null
╠══════════════════════════════════════════════════╣ THEN THROWS
Error: Value must be zero, got: Infinity
╚══════════════════════════════════════════════════╝"
`;

exports[`from > from(null) > from(null) 2`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
null
╠══════════════════════════════════════════════════╣ THEN THROWS
Error: Value must be zero, got: NaN
╚══════════════════════════════════════════════════╝"
`;

exports[`is > is("0") > is("0") 1`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
"0"
╠══════════════════════════════════════════════════╣ THEN RETURNS
false
╚══════════════════════════════════════════════════╝"
`;

exports[`is > is() > is() 1`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS

╠══════════════════════════════════════════════════╣ THEN RETURNS
false
╚══════════════════════════════════════════════════╝"
`;

exports[`is > is(-0.1) > is(-0.1) 1`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
-0.1
╠══════════════════════════════════════════════════╣ THEN RETURNS
false
╚══════════════════════════════════════════════════╝"
`;

exports[`is > is(-1) > is(-1) 1`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
-1
╠══════════════════════════════════════════════════╣ THEN RETURNS
false
╚══════════════════════════════════════════════════╝"
`;

exports[`is > is(0) > is(0) 1`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
0
╠══════════════════════════════════════════════════╣ THEN RETURNS
true
╚══════════════════════════════════════════════════╝"
`;

exports[`is > is(0) > is(0) 2`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
0
╠══════════════════════════════════════════════════╣ THEN RETURNS
true
╚══════════════════════════════════════════════════╝"
`;

exports[`is > is(0.1) > is(0.1) 1`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
0.1
╠══════════════════════════════════════════════════╣ THEN RETURNS
false
╚══════════════════════════════════════════════════╝"
`;

exports[`is > is(1) > is(1) 1`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
1
╠══════════════════════════════════════════════════╣ THEN RETURNS
false
╚══════════════════════════════════════════════════╝"
`;

exports[`is > is(null) > is(null) 1`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
null
╠══════════════════════════════════════════════════╣ THEN RETURNS
false
╚══════════════════════════════════════════════════╝"
`;

exports[`is > is(null) > is(null) 2`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
null
╠══════════════════════════════════════════════════╣ THEN RETURNS
false
╚══════════════════════════════════════════════════╝"
`;

exports[`is > is(null) > is(null) 3`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
null
╠══════════════════════════════════════════════════╣ THEN RETURNS
false
╚══════════════════════════════════════════════════╝"
`;

exports[`is > is(null) > is(null) 4`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
null
╠══════════════════════════════════════════════════╣ THEN RETURNS
false
╚══════════════════════════════════════════════════╝"
`;

exports[`tryFrom > tryFrom(-0.1) > tryFrom(-0.1) 1`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
-0.1
╠══════════════════════════════════════════════════╣ THEN RETURNS
null
╚══════════════════════════════════════════════════╝"
`;

exports[`tryFrom > tryFrom(-1) > tryFrom(-1) 1`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
-1
╠══════════════════════════════════════════════════╣ THEN RETURNS
null
╚══════════════════════════════════════════════════╝"
`;

exports[`tryFrom > tryFrom(0) > tryFrom(0) 1`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
0
╠══════════════════════════════════════════════════╣ THEN RETURNS
0
╚══════════════════════════════════════════════════╝"
`;

exports[`tryFrom > tryFrom(0) > tryFrom(0) 2`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
0
╠══════════════════════════════════════════════════╣ THEN RETURNS
0
╚══════════════════════════════════════════════════╝"
`;

exports[`tryFrom > tryFrom(0.1) > tryFrom(0.1) 1`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
0.1
╠══════════════════════════════════════════════════╣ THEN RETURNS
null
╚══════════════════════════════════════════════════╝"
`;

exports[`tryFrom > tryFrom(1) > tryFrom(1) 1`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
1
╠══════════════════════════════════════════════════╣ THEN RETURNS
null
╚══════════════════════════════════════════════════╝"
`;

exports[`tryFrom > tryFrom(null) > tryFrom(null) 1`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
null
╠══════════════════════════════════════════════════╣ THEN RETURNS
null
╚══════════════════════════════════════════════════╝"
`;

exports[`tryFrom > tryFrom(null) > tryFrom(null) 2`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
null
╠══════════════════════════════════════════════════╣ THEN RETURNS
null
╚══════════════════════════════════════════════════╝"
`;
